state/sqlite/sqlite_dbaccess.go

```diff
@@ -382,22 +389,20 @@ func (a *sqliteDBAccess) doSet(parentCtx context.Context, db querier, req *state
 				key = ?
 				AND etag = ?
 				AND (expiration_time IS NULL OR expiration_time > CURRENT_TIMESTAMP)`
+		var res sql.Result
 		res, err = db.ExecContext(ctx, stmt, requestValue, newEtag, isBinary, req.Key, *req.ETag)
-	}
-	if err != nil {
-		return err
-	}
+		if err != nil {
+			return err
+		}
 
-	// Count the number of affected rows
-	rows, err := res.RowsAffected()
-	if err != nil {
-		return err
-	}
-	if rows == 0 {
-		if req.HasETag() || req.Options.Concurrency == state.FirstWrite {
+		rows, err := res.RowsAffected()
+		if err != nil {
+			return err
+		}
+		if rows == 0 {
+			// 0 affected rows means etag failure
```

Comment:
Won't an update statement that, for example, tries to change an existing value to itself also return 0 rows affected?

Reply:
Looks like no:

```
sqlite> select * from test;
mykey9|myvalue9
sqlite> update test set value = "myvalue9" where key = "mykey9";
sqlite> select changes()
1
```

Reply:
Ok. For MySQL it will, but this is relevant here only to SQLite.